tests/cached.rs

```diff
@@ -294,12 +294,16 @@ fn test_racing_duplicate_keys_do_not_duplicate_sized_cache_ordering() {
     slow_small_cache("g", "h");
 }
 
+// String is not cloneable. So this also tests that the Result type
```

Comment:
`String` is cloneable. You could introduce a `struct NoClone` and return `Err(NoClone{})` to test this

Reply:
Not sure how I missed that. Wow.

And yup, that seems like a good solution.